lib/spack/spack/cmd/help.py

```diff
@@ -91,7 +91,7 @@ def setup_parser(subparser):
     help_spec_group = subparser.add_mutually_exclusive_group()
     help_spec_group.add_argument(
         '--spec', action='store_const', dest='guide', const='spec',
-        default=None, help='print all available commands')
+        default=None, help='print guide to specs')
```

Comment:
`spack help` lists:
```
more help:
  spack help --all       list all commands and options
  spack help <command>   help on a specific command
  spack help --spec      help on the spec syntax
  spack docs             open http://spack.rtfd.io/ in a browser
```
while `spack help help` lists:
```
optional arguments:
  -h, --help    show this help message and exit
  -a, --all     print all available commands
  --spec        print all available commands
```
Let's make the messages for `--all` and `--spec` match.